README.md
# Docker container images with "headless" VNC session

This repository contains Synchrotron SOLEIL's Docker Image. This is heavily based on Consol's Docker images.

The Docker image is installed with the following components:

* Desktop environment [**Xfce4**](http://www.xfce.org)
* VNC-Server (default VNC port `5901`)
* [**noVNC**](https://github.com/novnc/noVNC) - HTML5 VNC client (default http port `6901`)
* Browsers:
  * Mozilla Firefox
  * Chromium
* Foxtrot (SAXS datareduction software)
* Global Reducer (Tomography datareduction software)
* Orange

![Docker VNC Desktop access via HTML page](.pics/vnc_container_view.png)

## Build Status
![Docker Automated build](https://img.shields.io/docker/automated/gregviguier/centos-xfce-vnc-soleil.svg)



## OpenShift / Kubernetes

It's also possible to run the images in container orchestration platforms like [Kubernetes](https://kubernetes.io) or [OpenShift](https://openshift.io/). For more information how to deploy containers in the cluster, take a look at:

* [Kubernetes usage of "headless" VNC Docker images](./kubernetes/README.md)
* [OpenShift usage of "headless" VNC Docker images](./openshift/README.md) 

## Usage
Usage is **similar** for all provided images, e.g. for `consol/centos-xfce-vnc`:

- Print out help page:

      docker run consol/centos-xfce-vnc --help

- Run command with mapping to local port `5901` (vnc protocol) and `6901` (vnc web access):

      docker run -d -p 5901:5901 -p 6901:6901 gregviguier/centos-xfce-vnc-soleil
  
- Change the default user and group within a container to your own with adding `--user $(id -u):$(id -g)`:

      docker run -d -p 5901:5901 -p 6901:6901 --user $(id -u):$(id -g) gregviguier/centos-xfce-vnc-soleil

- If you want to get into the container use interactive mode `-it` and `bash`
      
      docker run -it -p 5901:5901 -p 6901:6901 gregviguier/centos-xfce-vnc-soleil bash


# Connect & Control
If the container is started like mentioned above, connect via one of these options:

* connect via __VNC viewer `localhost:5901`__, default password: `vncpassword`
* connect via __noVNC HTML5 full client__: [`http://localhost:6901/vnc.html`](http://localhost:6901/vnc.html), default password: `vncpassword` 
* connect via __noVNC HTML5 lite client__: [`http://localhost:6901/?password=vncpassword`](http://localhost:6901/?password=vncpassword) 


## Hints

### 1) Extend a Image with your own software
Since version `1.1.0` all images run as non-root user per default, so if you want to extend the image and install software, you have to switch back to the `root` user:

```bash
## Custom Dockerfile
FROM consol/centos-xfce-vnc
ENV REFRESHED_AT 2018-03-18

# Switch to root user to install additional software
USER 0

## Install a gedit
RUN yum install -y gedit \
    && yum clean all

## switch back to default user
USER 1000
```

### 2) Change User of running Container

Per default, since version `1.3.0` all container processes will be executed with user id `1000`. You can change the user id as follows: 

#### 2.1) Using root (user id `0`)
Add the `--user` flag to your docker run command:

    docker run -it --user 0 -p 6911:6901 gregviguier/centos-xfce-vnc-soleil

#### 2.2) Using user and group id of host system
Add the `--user` flag to your docker run command:

    docker run -it -p 6911:6901 --user $(id -u):$(id -g) gregviguier/centos-xfce-vnc-soleil

### 3) Override VNC environment variables
The following VNC environment variables can be overwritten at the `docker run` phase to customize your desktop environment inside the container:
* `VNC_COL_DEPTH`, default: `24`
* `VNC_RESOLUTION`, default: `1280x1024`
* `VNC_PW`, default: `my-pw`

#### 3.1) Example: Override the VNC password
Simply overwrite the value of the environment variable `VNC_PW`. For example in
the docker run command:

    docker run -it -p 5901:5901 -p 6901:6901 -e VNC_PW=my-pw gregviguier/centos-xfce-vnc-soleil

#### 3.2) Example: Override the VNC resolution
Simply overwrite the value of the environment variable `VNC_RESOLUTION`. For example in
the docker run command:

    docker run -it -p 5901:5901 -p 6901:6901 -e VNC_RESOLUTION=800x600 gregviguier/centos-xfce-vnc-soleil
    
### 4) View only VNC
Since version `1.2.0` it's possible to prevent unwanted control via VNC. Therefore you can set the environment variable `VNC_VIEW_ONLY=true`. If set, the startup script will create a random password for the control connection and use the value of `VNC_PW` for view only connection over the VNC connection.

     docker run -it -p 5901:5901 -p 6901:6901 -e VNC_VIEW_ONLY=true gregviguier/centos-xfce-vnc-soleil


## Contributors

At this point we want to thank all contributors, which helped to move this great project by submitting code, writing documentation, or adapting other tools to play well together with the docker headless container.

* [Tobias Schneck](https://github.com/toschneck) - Lead development
* [Robert Bohne](https://github.com/rbo) - IceWM images
* [hsiaoyi0504](https://github.com/hsiaoyi0504) - PR [#66](https://github.com/ConSol/docker-headless-vnc-container/pull/66)
* [dmhumph](https://github.com/dmhumph) - PR [#44](https://github.com/ConSol/docker-headless-vnc-container/issue/44) 
* [Simon Hofmann](https://github.com/s1hofmann) - Current console maintainer


Foxtrot-3.5.2/RELEASE_NOTES.txt
-------------------------------------------
 Foxtrot RELEASE NOTES
-------------------------------------------

Format:
================================
Version (Date)
================================
 - SOLEIL Jira reference(s): description

-------------------------------------------


================================
3.5.2-3645 (February 2019)
================================
- DATAREDUC-845: Improvement of CAHPLCSample algorithm
- DATAREDUC-852: Bug fix: Filtered Input are not extracted for new CA
- DATAREDUC-843: Import speed improvement
- DATAREDUC-797, DATAREDUC-860: Load ctx data file in macro
- DATAREDUC-861: Missing paths in last used paths file
- DATAREDUC-859: Bug fix: Calibration process regression + remove automatic override
- DATAREDUC-855: Bug fix: Importation of foxtrot output files is not working
- DATAREDUC-848: Make autoQRange option of CAHPLCSample user macro compatible with defaultUserMacro file
- DATAREDUC-841: Bug fix: Foxtrot too slow with nexus files in multi threads mode.
- DATAREDUC-847: Bug fix: Missing chart formulas
- DATAREDUC-848: Add autoQRange in CAHPLCSample preferences
- DATAREDUC-849: Add hkl files in Foxtrot distribution
- DATAREDUC-853: Bug fix: Used memory not perfectly cleaned
- DATAREDUC-819: Bug fix: Waiting dialog during export disappeared, though export was still going on)
- DATAREDUC-858: Add some persisting data in Foxtrot dictionaries
- DATAREDUC-862: Foxtrot properties files content updated

================================
3.5.1-3613 (December 2018)
================================
- DATAREDUC-801: Dynamic center for sector calculation
- Bug fix: Data export bug due to undesired commit

================================
3.5.0-3606 (November 2018)
================================
- DATAREDUC-797: Load of context data file in macro
- DATAREDUC-822: Remember last used paths
- DATAREDUC-799: Add new "Divide" userMacro
- DATAREDUC-813: Bug fix: Sigma calculation not saved for LinearFunction macro
- DATAREDUC-823: Alternative sigma calculation in circleGathering
- JAVAAPI-440: Bug fix: Intensity monitor reading error due to NullPointerException
- DATAREDUC-790: Real export of angular sector parameters
- DATAREDUC-839: Bug fix: Extraction misses some files
- DATAREDUC-724: Change clever average criteria
- JAVAAPI-456: Add the possibility to customize curves number format in chart
- DATAREDUC-752: Extraction takes too much time
- DATAREDUC-806: Change "show selected data" icon
- DATAREDUC-719, DATAREDUC-676: CleverAverage5: generate fit results
- DATAREDUC-821: Change I0_ref selection algorithm in clever average 5 bis
- DATAREDUC-826: Bug fix: Import failed

================================
3.4.9-3471 (April 2018)
================================
- DATAREDUC-788: Freeze of Foxtrot with long path in Edit Directory
- DATAREDUC-789: Add the possibility to hide Set Input button in new Sum userMacro
- DATAREDUC-787: Wrong context data transmission in Azimuthal Integration
- CleverAverage5 compatibility ensured in python extractor

================================
3.4.8-3447 (February 2018)
================================
- DATAREDUC-776: Context data format after calibration
- DATAREDUC-777: SumSpectra macro full java implementation
- DATAREDUC-778: MergeOnQ_NoResampling failed to execute
- DATAREDUC-781: Better NaN management in projections
- DATAREDUC-782: Keep mono-thread mode by default for xenocs version
- DATAREDUC-768: Persisting data format management.
  For this case, you should setup a "format" attribute at the end of the key mapping in the mapping dictionary.
  You should use, as value "%.Xf" for float format or "%.Xe" for scientific format, where "X" is the desired number of decimal digits.
  Example in a nexus mapping dictionary:
  <item key="MyData">
    <path>/{NXdata}/my_data</path>
    <call>HarvestEquipmentAttributes</call>
    <attribute name="format" value="%.3f"/>
  </item>
- DATAREDUC-770: Persisting data randomly not displayed in fit table
- DATAREDUC-772: Indicate which plugin is not multi-threads compatible
- DATAREDUC-779: Block Azimuthal Integration for pixelStepValue !=1


================================
3.4.7-3399 (February 2018)
================================
- DATAREDUC-756: Change calibration highlight color
- DATAREDUC-765: Change default value for initial guesses in calibration
- DATAREDUC-745: Color scale is lost sometimes
- Calibration highlighting bug correction (wrong delta recovering for highlighted pixels: calibration was OK but not the highlighting)


================================
3.4.6-3390 (February 2018)
================================
- DATAREDUC-726: Precision step in Azimuthal Integration
- DATAREDUC-727: Bug fix: Projection failed
- DATAREDUC-737: Bug fix: Quadratic relative difference of calibration not refreshed
- DATAREDUC-738: Bug fix: Wrong parameters in Azimuthal Integration
- DATAREDUC-750: Bug fix: Missing pixelSplitting and xValuesRefined options of Azimuthal Integration in macro file
- DATAREDUC-751: Bug fix: Wrong update of data chart when load data file
- DATAREDUC-728: Improvement of calibration ui
- DATAREDUC-736: Specific highlighter for calibration ui
- DATAREDUC-739: Bug fix: Wrong title file for imported results
- DATAREDUC-740: Bug fix: Wrong result name operations for Averaged user macros
- DATAREDUC-741: Bug fix: Scaling Macro failed
- DATAREDUC-743: AzimuthalIntegration time process is too slow
- DATAREDUC-744: Bug fix: Import nxs file loading panel disappear before loading process is done
- CTRLRFC-738, DATAREDUC-746: Waited average algorithm reviewed in case of negative values
- DATAREDUC-749: Bug fix: Nexus files reading impossible after about 2 minutes
- DATAREDUC-748: Customize nexus files lock lifetime
- DATAREDUC-753: Sigma calculation algorithm reviewed

Foxtrot-3.5.2/resources/tools/add.sh
#!/bin/sh

#---------------------------------------------------
#- Revision: 1.1                                   -
#- Author: R. GIRARDOT                             -
#- Date: December 01 2017                          -
#- Purpose: Launch 2D Operation                    -
#---------------------------------------------------

#--------------------------------
#- 2D operation Root Java Paths -
#--------------------------------
cd `dirname $0`
ROOT_DIR=`pwd`
cd -
cd $ROOT_DIR/../..
TOP_DIR=`pwd`
cd -
JAR_ROOT=$TOP_DIR/lib

#---------------------
#- Test Java version -
#---------------------
# Expected Java version
JAVA_VERSION=7
# Installed Java version: since Java 9 -> version.X.Y
JAVA_INSTALLED_VERSION=`java -version 2>&1 | head -n 1 | awk -F '"' '{print $2}'| cut -d'.' -f1`
if [ "$JAVA_INSTALLED_VERSION" = "1" ]
then
# Installed Java version: before Java 9 -> 1.version.X
JAVA_INSTALLED_VERSION=`java -version 2>&1 | head -n 1 | awk -F '"' '{print $2}'| cut -d'.' -f2`
fi
if [ "$JAVA_INSTALLED_VERSION" = "" ]
then
# Java is not even installed
echo "***********************************"
echo "No java installed!"
echo "Please install java $JAVA_VERSION or newer"
echo "***********************************"
elif [ $((JAVA_INSTALLED_VERSION-$JAVA_VERSION)) -lt 0 ]
then
# Too old version of Java
echo "***********************************"
echo "No valid java version found."
echo "You have Java $JAVA_INSTALLED_VERSION installed."
echo "This software needs java $JAVA_VERSION or newer."
echo "***********************************"
else
# Java version OK

#--------------------------
#- SOURCE COMMON ENV VARS -
#--------------------------
CLASSPATH=
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.gui-image-sum-and-mean-1.2.2-3627.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.gui-imageoperation-commons-1.2.2-3627.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/ch.qos.logback-logback-classic-1.2.3.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-CDMABox-1.0.1-7026.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.cdma-core-3.4.5-2250.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.cdma.plugin-nexus-soleil-3.4.5-2250.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.cdma.plugin-edf-soleil-3.4.5-2250.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-BasicUtilities-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib.comete-CometeDefinition-2.5.1-5756.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/ch.qos.logback-logback-core-1.2.3.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.slf4j-slf4j-api-1.7.25.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-SwingUtilities-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib.comete-CometeSwing-2.6.4-5814.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib.comete-CometeBox-1.2.3-7236.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.cdma-utilities-3.4.5-2250.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib.comete-CDMADataSource-1.2.2-5719.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/com.famfamfam-silk-1.3.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.tango-project-tango-icon-theme-0.8.0.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.apache.felix-org.osgi.core-1.4.0.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.jdom-jdom2-2.0.5.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.cdma.engine-hdf-3.4.5-2250.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/commons-io-commons-io-2.0.1.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-DataConnectionManagement-1.3.2-5816.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-AwtUtilities-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/com.miglayout-miglayout-3.7.4.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-SwingXUtilities-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-ExpressionAbstraction-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-ExpressionJEval-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-ImageJUtilities-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib.comete-CometeAWT-2.1.11-5689.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/commons-lang-commons-lang-2.4.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.swinglabs-colorchooser-0.0.1.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.swinglabs-swingx-1.6.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/gov.nih-ij-1.45s.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/ncsa.hdf-h5object-2.10.0.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/ncsa.hdf-hdf5lib-2.10.0.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/ncsa.hdf-object-2.10.0.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.apache.commons-commons-math3-3.5.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/com.jhlabs-filters-2.0.235.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.swinglabs-swing-worker-1.1.jar


#--------------
#- VM Options -
#--------------
JAVA_OPTIONS="-Xms256m -Xmx1024m -Dcom.sun.management.jmxremote=true"


#-------------------
#- USER PARAMETERS -
#-------------------
CDMA_DICTIONARY_ROOT=$TOP_DIR/CDMA
JAVA_BITS=`java -version 2>&1 | grep 64-Bit`
if [ "$JAVA_BITS" = "" ]
then
# 32bits JVM
NEXUS_PATH=$TOP_DIR/native/linux32
else
# 64bits JVM
NEXUS_PATH=$TOP_DIR/native/linux64
fi

#--------------------------
#- Start the 2D operation -
#--------------------------
java $JAVA_OPTIONS -cp $CLASSPATH \
-Djava.library.path=$NEXUS_PATH \
-Duser.language=en \
-Duser.country=US \
-Dlogback.configurationFile=$ROOT_DIR/conf/imageoperation-logback.xml \
-DCDMA_RESOURCES_DIRECTORY=$CDMA_DICTIONARY_ROOT \
fr.soleil.image.sumandmean.ImageSum $@
fi


Foxtrot-3.5.2/foxtrot.sh
#!/bin/sh

#---------------------------------------------------
#- Revision: 3.20                                  -
#- Author: R. GIRARDOT / G. VIGUIER                -
#- Date: February 11 2019                          -
#- Purpose: Launch Foxtrot                         -
#---------------------------------------------------
echo "Checking environment..."

#---------------------------------
#- DataReduction Root Java Paths -
#---------------------------------
cd `dirname $0`
ROOT_DIR=`pwd`
cd -
JAR_ROOT=$ROOT_DIR/lib

#---------------------
#- Test Java version -
#---------------------
echo "Checking Java version..."
# Expected Java version
JAVA_VERSION=7
# Installed Java version: since Java 9 -> version.X.Y
JAVA_INSTALLED_VERSION=`java -version 2>&1 | head -n 1 | awk -F '"' '{print $2}'| cut -d'.' -f1`
if [ "$JAVA_INSTALLED_VERSION" = "1" ]
then
# Installed Java version: before Java 9 -> 1.version.X
JAVA_INSTALLED_VERSION=`java -version 2>&1 | head -n 1 | awk -F '"' '{print $2}'| cut -d'.' -f2`
fi
if [ "$JAVA_INSTALLED_VERSION" = "" ]
then
# Java is not even installed
echo "***********************************"
echo "No java installed!"
echo "Please install java $JAVA_VERSION or newer"
echo "***********************************"
elif [ $((JAVA_INSTALLED_VERSION-$JAVA_VERSION)) -lt 0 ]
then
# Too old version of Java
echo "***********************************"
echo "No valid java version found."
echo "You have Java $JAVA_INSTALLED_VERSION installed."
echo "This software needs java $JAVA_VERSION or newer."
echo "***********************************"
else
# Java version OK
echo "Setting up Java..."
#--------------------------
#- SOURCE COMMON ENV VARS -
#--------------------------
CLASSPATH=
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.gui-foxtrot-3.5.2-3650.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.gui-imagereducer-3.5.2-3650.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-DockingVl-1.1.1-5570.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/com.famfamfam-silk-1.3.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.tango-project-tango-icon-theme-0.8.0.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.swinglabs-colorchooser-0.0.1.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-BasicUtilities-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-SwingUtilities-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib.comete-CometeDefinition-2.5.1-5756.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib.comete-CometeSwing-2.6.4-5814.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-NumericalComputing-2.2.0-3647.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib.ij.macro-MacroManager-0.1.0.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.gui-reduction-tools-3.5.2-3650.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.gui-data-export-3.5.2-3650.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-DockingCore-1.1.2-5570.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/commons-collections-commons-collections-3.2.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.apache.commons-commons-math3-3.5.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/com.vlsolutions-vldocking-3.0.2.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-AwtUtilities-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/com.miglayout-miglayout-3.7.4.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-DataConnectionManagement-1.3.2-5816.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-SwingXUtilities-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-ExpressionAbstraction-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-ExpressionJEval-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-ImageJUtilities-1.3.6-5812.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib.comete-CometeAWT-2.1.11-5689.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/commons-lang-commons-lang-2.4.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.swinglabs-swingx-1.6.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/gov.nih-ij-1.45s.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.apache.maven-maven-plugin-api-2.0.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib.comete-CometeBox-1.2.3-7236.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.cdma-core-3.4.5-2250.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.cdma.plugin-nexus-soleil-3.4.5-2250.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.cdma.plugin-edf-soleil-3.4.5-2250.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.cdma-utilities-3.4.5-2250.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib.comete-CDMADataSource-1.2.2-5719.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/fr.soleil.lib-CDMABox-1.0.1-7026.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/com.jhlabs-filters-2.0.235.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.swinglabs-swing-worker-1.1.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.apache.felix-org.osgi.core-1.4.0.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.jdom-jdom2-2.0.5.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.cdma.engine-hdf-3.4.5-2250.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/commons-io-commons-io-2.0.1.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.slf4j-slf4j-simple-1.7.7.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/ncsa.hdf-h5object-2.10.0.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/ncsa.hdf-hdf5lib-2.10.0.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/ncsa.hdf-object-2.10.0.jar
CLASSPATH=$CLASSPATH:$JAR_ROOT/org.slf4j-slf4j-api-1.7.25.jar

#----------------
#- CRASH REPORT -
#----------------
BASE_FILE=$ROOT_DIR/resources/dumps/Foxtrot-$(date +%y%m%d-%H%M%S) 
JVM_DUMP_OPTIONS="-XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=$BASE_FILE.hprof -XX:ErrorFile=$BASE_FILE.log"

#----------------------------------
#- USER PARAMETERS and VM Options -
#----------------------------------
JAVA_BITS=`java -version 2>&1 | grep 64-Bit`
if [ "$JAVA_BITS" = "" ]
then
# 32bits JVM
JAVA_OPTIONS="-Xms256m -Xmx1024m -Dcom.sun.management.jmxremote=true"
NEXUS_PATH=$ROOT_DIR/native/linux32
else
# 64bits JVM
JAVA_OPTIONS="-Xms256m -Xmx2048m -Dcom.sun.management.jmxremote=true"
NEXUS_PATH=$ROOT_DIR/native/linux64
fi
CDMA_DICTIONARY_ROOT=$ROOT_DIR/CDMA
PERSPECTIVEDIRECTORY=$ROOT_DIR/resources/perspective
WORKINGPATH=$HOME
FITDIRECTORY=$ROOT_DIR/resources/fit
FORMULASDIRECTORY=$ROOT_DIR/resources/axisMacro
MASKDIRECTORY=$ROOT_DIR/resources/masks
IMAGEMACRODIRECTORY=$ROOT_DIR/resources/macros
USERMACRODIRECTORY=$ROOT_DIR/resources/userMacros
USERCONTEXTDATADIRECTORY=$ROOT_DIR
EXTERNALWORKERDIRECTORY=$ROOT_DIR/resources/tools
LASTUSEDPATHSFILE=$ROOT_DIR/resources/paths/lastUsedPaths-linux.properties
EXPORTPATH=$HOME
EXTRACTORPATH=$ROOT_DIR/python/extractor.sh
USERGUIDE=$ROOT_DIR/doc/Foxtrot-UserGuide.pdf
# -------------------------------------
# - Calibration highlight color setup -
# -------------------------------------
# CALIBRATION_COLOR represents the highlight color you want to use in detector calibration.
# It is expected to be encoded one of these 2 ways:
# CALIBRATION_COLOR=R,G,B
# or
# CALIBRATION_COLOR=R,G,B,A
# where:
# R,G,B and A are integers between 0 and 255 included.
# R represents red, G green, B blue, and A alpha.
# For alpha, 0 means totally transparent (i.e. invisible) and 255 means totally opaque (i.e. you won't see any pixel color under it).
# If not set, default alpha value will be 120.
# If not set or badly configured, CALIBRATION_COLOR will take the value "0,0,0,120".
# 245,159,200 is pink
CALIBRATION_COLOR=245,159,200,100
# ------------------------------------
# - Nexus files locks cleaning setup -
# ------------------------------------
# NEXUS_ACCESS_CHECK_PERIOD represents at which frequency nexus files locks are checked for their lifetime.
# If not correctly set, NEXUS_ACCESS_CHECK_PERIOD will be set by default to 30 seconds.
# NEXUS_LOCK_LIFETIME represents the lifetime of nexus files locks.
# If a nexus file last access is older than NEXUS_LOCK_LIFETIME, then the associated lock, granted by the OS, will be cleared.
# If not correctly set, NEXUS_LOCK_LIFETIME will be set by default to 2 minutes.
# The 2 properties NEXUS_ACCESS_CHECK_PERIOD and NEXUS_LOCK_LIFETIME follow this logic:
# ValueUnit, where Value is a strictly positive integer, and Unit a time unit.
# Unit can be any of these: milliseconds, seconds, minutes, hours.
# The remaining 's' (for plural) is optional, i.e. Unit can also be millisecond, second, minute, hour.
# Unit is optional. If not set, Value represents milliseconds.
# Examples (1 example per line):
# 40000
# 60000milliseconds
# 20seconds
# 1hour
# 3minutes
NEXUS_ACCESS_CHECK_PERIOD=30seconds
NEXUS_LOCK_LIFETIME=2minutes

#--------------------------------
#- Start Foxtrot                -
#--------------------------------
echo "Launching Foxtrot..."
java $JVM_DUMP_OPTIONS $JAVA_OPTIONS -cp $CLASSPATH \
-Djava.library.path=$NEXUS_PATH \
-Dswing.defaultlaf=javax.swing.plaf.metal.MetalLookAndFeel \
-DCDMA_RESOURCES_DIRECTORY=$CDMA_DICTIONARY_ROOT \
-DNexusAccessCheckPeriod=$NEXUS_ACCESS_CHECK_PERIOD \
-DNexusLockLifetime=$NEXUS_LOCK_LIFETIME \
fr.soleil.fusion.image.Foxtrot \
WorkingPath=$WORKINGPATH \
ExportPath=$EXPORTPATH \
ExtractorPath=$EXTRACTORPATH \
ImageMacroDirectory=$IMAGEMACRODIRECTORY \
UserMacroDirectory=$USERMACRODIRECTORY \
UserContextDataDirectory=$USERCONTEXTDATADIRECTORY \
MaskDirectory=$MASKDIRECTORY \
PerspectiveDirectory=$PERSPECTIVEDIRECTORY \
FitDirectory=$FITDIRECTORY \
FormulasDirectory=$FORMULASDIRECTORY \
ExternalWorkerDirectory=$EXTERNALWORKERDIRECTORY \
LastUsedPathsFile=$LASTUSEDPATHSFILE \
UserGuide=$USERGUIDE \
ContextModificationInMacrosEnabled=true \
DisplayImageLoading=false \
CalibrationColor=$CALIBRATION_COLOR \
FitEnabled=true \
ColorsEnabled=false \
Language=EN \
InnerRoiForDynamicMacro=true
fi
